### PR TITLE
Added span first, span near, span no, spar or queries

### PR DESCRIPTION
**Pull request by chyzas**

### CODE REVIEW

File: DSL/Query/SpanNearQuery.php [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+    /**
+     * @param array $queries
+     * @param array $parameters
+     *
+     * @throws \LogicException
+     */
+    public function __construct(array $queries = [], array $parameters = [])
+    {
+        foreach ($queries as $query) {
```
Comment (martiis) [inline DSL/Query/SpanNearQuery.php]: Couldn't just `$this->queries = $queries`?

File: DSL/Query/SpanNearQuery.php [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+{
+    use ParametersTrait;
+
+    /**
+     * @var BuilderInterface[]
+     */
+    private $queries = [];
+
+    /**
+     * @param array $queries
```
Comment (martiis) [inline DSL/Query/SpanNearQuery.php]: not an `array` but `BuilderInterface[]`

File: DSL/Query/SpanFirstQuery.php [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+     * @param BuilderInterface $query
+     * @param array            $parameters
+     *
+     * @throws \LogicException
+     */
+    public function __construct($query, array $parameters = [])
+    {
+        $this->query = $query;
+        $this->setParameters($parameters);
+        if (!$this->hasParameter('end')) {
```
Comment (saimaz) [inline DSL/Query/SpanFirstQuery.php]: `end` is mandatory field so it must be in the constructor.

File: DSL/Query/SpanFirstQuery.php [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+     */
+    private $query;
+
+    /**
+     * @param BuilderInterface $query
+     * @param array            $parameters
+     *
+     * @throws \LogicException
+     */
+    public function __construct($query, array $parameters = [])
```
Comment (saimaz) [inline DSL/Query/SpanFirstQuery.php]: `$query` must have `BuilderInterface` type hint

**Pull request closed without merging**